### DISCUSSION TITLE
Business Manager – get list of all content assets, whether or not assigned to folder

### DISCUSSION THREAD

**Discussion by GeneralPatten** [SalesforceCommerceCloud/sfcc-ci#533] [Team Posts]
This is a question that has been been vexing me for some time now, and I have yet to find a solution, even after having spoken with CC platform developers at Connections last year.

For a custom Business Manager module, we would like to get a list of **all** content assets for the site library — including those not assigned to a folder, not marked as online, and not marked as searchable — similar to the Business Manager Content Assets module, or the Content Assets dialog from the Content Slot configuration page. However, it seems that this ability is not exposed, at least publicly, in any of the SFCC APIs.

- ContentSearchModel is not an option since A) it does not support wildcards and B) offline and unsearchable assets would be excluded
- Recursively retrieving all content assets from root and subfolders also does not work because many content assets are never assigned to a folder
- OCAPI API methods also have the same issues as the two options above

One option would be to leverage the request to _Slot-ContentAssetsJSON_ used by the Content Assets dialog during Content Slot configuration. However, only the ID and display name are returned from this request (the JSON returned is invalid as well, but this can be worked around) and we would like to be able to include additional fields. Of course, there is the more important fact that using Slot-ContentAssetsJSON is unsupported and poses a risk if it's ever changed or removed.

Obviously, based on the call to Slot-ContentAssetsJSON, there must be a way to retrieve this data. It simply seems that it's not exposed in the public API.

Thoughts?

**Comment by mihaivilcuosf** [comment]
you can export the library and extract the ids from there
